File: Unit2-02/Typescript/MrCoxallStack.ts
/**
* This is the MrCoxallStack class
*
* By:      Evan Beaudoin
* Version: 1.0
* Since:   2024-03-25
*/ 

export class MrCoxallStack {
  // Stack that contains all of the string
  private _stackAsList: string[]

  // Creates the stack
  public constructor() {
    this._stackAsList = []
  }

  // Getter for stackAsList
  get stackAsList() {
    return this._stackAsList;
  }

  // Setter for stackAsList
  set stackAsList(stack: string[]) {
    this._stackAsList = stack;
  }

  // Pushes the values to the stack
  public push(stringPushed) {
    this._stackAsList.push(stringPushed)
  }

  // Returns the stack
  public showStack() {
    let stackItems = ""
    
    for (let counter = 0; counter < this._stackAsList.length; counter++) {
      // If statments ensures that comma is not printed at the start
      if (counter === 0) {
        stackItems += this._stackAsList[counter]
      } else {
        stackItems += ", " + this._stackAsList[counter]
      }
    }

    return stackItems
  }

  public popItem() {
    let returnValue = ""

    if (this._stackAsList.length > 0) {
      returnValue = this._stackAsList.pop()
    }
    else {
      console.log("Array is empty!")
    }
    return returnValue
  }
}

File: Unit2-02/Java/MrCoxallStack.java
/*
* This class defines a string stack
*
* @author  Evan Beaudoin
* @version 1.0
* @since   2024-03-25
*/

import java.util.List;

/**
 * This is the class.
*/
public class MrCoxallStack {
    /**
     * Adds strings to an array.
     *
     * @param array contains the array of strings
    */
    private List<String> stackAsList;

    /**
<<<<<<< HEAD
    * Returns the stack value.
    *
    * @return the values of the list formatted
=======
     * Constructs a new instance of MrCoxallStack.
     * Initializes the stack with an empty ArrayList.
     *
    */
    public MrCoxallStack() {
        this.stackAsList = new ArrayList<>();
    }

    /**
     * The getStackAsList() getter.
     *
     * @return -  current list
     */
    public List<String> getStackAsList() {
        return this.stackAsList;
    }

    /**
     * Pushes the string to a list.
     *
     * @param string the string that is going to be pushed
    */
    public void push(String string) {
        this.stackAsList.add(string);
    }

    /**
     * Returns the stack value.
     *
     * @return the values of the list formatted
>>>>>>> b49eb87def776a148bfccc38c3527cf0a1d36018
    */
    public String showStack() {
        final StringBuilder stackValues = new StringBuilder();
        // https://www.javatpoint.com/StringBuilder-class

        for (String value : this.stackAsList) {
            stackValues.append(value).append(", ");
        }

        stackValues.deleteCharAt(stackValues.length() - 2);

        return stackValues.toString();
    }

    /**
     * Pushes the string to a list.
     *
     * @param string the string that is going to be pushed
    */
    public void push(String string) {
        this.stackAsList.add(string);
    }

    /**
     * Returns the stack value.
     *
     * @return the values of the list formatted
     *
     * @throws IllegalStateException if this is ever called
    */
    public String popItem() {
        if (stackAsList.isEmpty()) {
            throw new IllegalStateException("Stack is empty.");
        }

        return stackAsList.remove(stackAsList.size() - 1);
    }
}
